internal/server/singleprocess/service_task_test.go

```diff
@@ -0,0 +1,369 @@
+package singleprocess
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/waypoint/pkg/server"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+	serverptypes "github.com/hashicorp/waypoint/pkg/server/ptypes"
+)
+
+func TestServiceTask(t *testing.T) {
+	ctx := context.Background()
+
+	// Create our server
+	impl, err := New(WithDB(testDB(t)))
+	require.NoError(t, err)
+	client := server.TestServer(t, impl)
+
+	type Req = pb.UpsertTaskRequest
+
+	t.Run("create and update", func(t *testing.T) {
+		require := require.New(t)
+
+		// Create, should get an ID back
+		resp, err := client.UpsertTask(ctx, &pb.UpsertTaskRequest{
+			Task: serverptypes.TestValidTask(t, &pb.Task{TaskJob: &pb.Ref_Job{Id: "run_job"}}),
+		})
+		require.NoError(err)
+		require.NotNil(resp)
+		result := resp.Task
+		require.NotEmpty(result.Id)
+
+		// Let's write some data
+		result.StartJob = &pb.Ref_Job{Id: "start_job"}
+		result.StopJob = &pb.Ref_Job{Id: "stop_job"}
+		resp, err = client.UpsertTask(ctx, &pb.UpsertTaskRequest{
+			Task: result,
+		})
+		require.NoError(err)
+		require.NotNil(resp)
+		result = resp.Task
+		require.Equal(result.StartJob.Id, "start_job")
```

Comment:
nit: let's test that this new `result.Id` is equal to the same ID on line ~35

Reply:
Sounds good! Got that updated locally, will push once I figure out the NotFound on Delete test